tracker/form/login.py

```diff
@@ -45,3 +52,28 @@ def fail():
             return False
         self.user = user
         return True
+
+
+class OTPForm(BaseForm):
+    otp_code = StringField(u'OTP Code', validators=[DataRequired(), Length(max=6)])
+    login = SubmitField(u'login')
+
+    def validate(self):
+        self.user = None
+        rv = BaseForm.validate(self)
+        if not rv:
+            return False
+
+        def fail():
+            self.otp_code.errors.append(ERROR_INVALID_OTP)
+            return False
+
+        if not session.get('two_factor'):
+            return False
+
+        user = User.query.filter(User.name == session['two_factor']).first()
```

Comment:
Please don't do any additional queries on the User object other then using `current_user` and call `db.session.commit()` if some database backed field was changed.